src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -734,37 +734,27 @@ private class FactoryContext
             public List<(ParameterExpression, Expression)> TryParseParams { get; } = new();
         }
 
-        private static class Log
+        private static partial class Log
         {
-            private static readonly Action<ILogger, Exception> _requestBodyIOException = LoggerMessage.Define(
-                LogLevel.Debug,
-                new EventId(1, "RequestBodyIOException"),
-                "Reading the request body failed with an IOException.");
-
-            private static readonly Action<ILogger, Exception> _requestBodyInvalidDataException = LoggerMessage.Define(
-                LogLevel.Debug,
-                new EventId(2, "RequestBodyInvalidDataException"),
-                "Reading the request body failed with an InvalidDataException.");
-
-            private static readonly Action<ILogger, string, string, string, Exception?> _parameterBindingFailed = LoggerMessage.Define<string, string, string>(
-                LogLevel.Debug,
-                new EventId(3, "ParamaterBindingFailed"),
-                @"Failed to bind parameter ""{ParameterType} {ParameterName}"" from ""{SourceValue}"".");
-
             public static void RequestBodyIOException(HttpContext httpContext, IOException exception)
-            {
-                _requestBodyIOException(GetLogger(httpContext), exception);
-            }
+                => RequestBodyIOException(GetLogger(httpContext), exception);
+
+            [LoggerMessage(1, LogLevel.Debug, "Reading the request body failed with an IOException.", EventName = "RequestBodyIOException")]
```

Comment:
Standardize on a shared LoggerEventIds class? Inline IDs are hard to keep unique when using tested log classes.

Reply:
Something like this:

```
partial class Log
{
   private const int RequestBodyIOExceptionId = 1;
   private const int RequestBodyInvalidDataExceptionId = 2;

  [LoggerMessage(RequestBodyIOExceptionId , LogLevel.Debug, "Reading the request body failed with an IOException.", EventName = "RequestBodyIOException")]
 ```
?

Reply:
Nevermind, I forgot that some assemblies use different categories for different classes so they don't need globally unique ids.

Reply:
I'd kinda like to get rid of the constants class.